### ISSUE TITLE
New analyzer: Casting enum to an incompatible enum

### ISSUE THREAD

**Issue by ronnygunawan** [ronnygunawan/analyzers#48]
```cs
enum X { A, B }
enum Y { I, J, K }

X x = (X)Y.J; // Warning: casting enum to an incompatible enum. Value 2 is missing from 'X'
Y y = (Y)X.A; // no warning because X is a subset of Y

int i = 0;
X x = (X)i; // Warning: do not cast int to enum directly
```

**Comment by ronnygunawan** [comment]
Related analyzer: Possible unintended enum casting
```cs
enum X {
    A = -1,
    B,
    C
}
enum Y {
    A = -1,
    B = 1,
    C = 0
}

X y = (X)Y.A; // Warning: possibly an unintended enum casting: 'B' has different values in 'X' (is 0) and in 'Y' (is 1)
```

**Comment by ronnygunawan** [comment]
Implemented in 1.0.27
